app/models/user.rb

```diff
@@ -55,4 +55,12 @@ def requester?
   def possible_projects_professional
     Project.unassigned.joins(skills: :users).where("user_skills.user_id = ?", self.id)
   end
+
+  def role_to_add
+    (Role.signup_roles - roles)
+  end
+
+  def add_role
```

Comment:
Yes, I like add_extra_role better. And yes that's exactly what these two methods are doing.